File: 1/trebuchet.py
import time

def get_code(line):
    numbers = [char for char in line if char.isdigit()]
    code = int(numbers[0] + numbers[-1])
    return code

def main():
    start_time = time.time()
    total_sum = 0

    with open("input.txt", "r") as file:
        for line in file:
            line = line.strip()
            code = get_code(line)
            total_sum += code

    print(total_sum)
    print(f"Execution time: {(time.time() - start_time) * 1000000:.0f} microseconds")

if __name__ == "__main__":
    main()

File: 1/trebuchet.js
const fs = require('fs');

// Read input file
const input = fs.readFileSync('input.txt', 'utf8');

// Function to get the calibration value from a line
function getCode(line) {
  const numbers = line.split('').filter(char => /\d/.test(char));
  const code = `${numbers[0]}${numbers[numbers.length - 1]}`;
  return parseInt(code, 10);
}

// Main function
function main() {
  const startTime = new Date();
  const lines = input.trim().split('\n');
  const sum = lines.reduce((acc, line) => acc + getCode(line), 0);
  console.log(sum);
  console.log(`Execution time: ${new Date() - startTime} ms`);
}

main();